src/hooks/useSourceApi.ts

```diff
@@ -1,105 +1,332 @@
+        variant: 'danger',
+        id: helpers.generateId()
+      });
+    },
+    [onAddAlert, t]
+  );
+
+  const addSources = useCallback(
+    async (payload: SourceType) => {
+      try {
+        const response = await apiCall(payload);
+        callbackSuccess(response);
+      } catch (error) {
+        if (isAxiosError(error)) {
+          return callbackError(error, payload.name);
+        }
+        if (!helpers.TEST_MODE) {
+          console.error(error);
+        }
+      }
+    },
+    [apiCall, callbackSuccess, callbackError]
+  );
+
+  return {
+    apiCall,
+    callbackError,
+    callbackSuccess,
+    addSources
   };
+};
 
-  /**
-   * Sets the source that is currently being edited.
-   *
-   * @param {SourceType} source - The source to be set as the one being edited.
-   */
-  const onEditSource = (source: SourceType) => {
-    setSourceBeingEdited(source);
+const useEditSourceApi = (onAddAlert: (alert: AlertProps) => void) => {
+  const { t } = useTranslation();
+
+  const apiCall = useCallback(
+    (payload: SourceType): Promise<AxiosResponse<SourceType>> =>
+      axios.put(`${process.env.REACT_APP_SOURCES_SERVICE}${payload.id}/`, payload),
+    []
+  );
+
+  const callbackSuccess = useCallback(
+    (response: AxiosResponse<SourceType>) => {
+      const successMessage = t('toast-notifications.description', {
+        context: 'add-source_hidden_edit',
+        name: response.data.name
+      });
+      onAddAlert({
+        title: successMessage,
+        variant: 'success',
+        id: helpers.generateId()
+      });
+
+      return;
+    },
+    [onAddAlert, t]
+  );
+
+  const callbackError = useCallback(
+    (error: AxiosError<ApiSourceErrorType>, name: string) => {
+      const errorMessage = t('toast-notifications.title', {
+        context: 'add-source_hidden_error_edit',
+        name: name,
+        message: apiHelpers.extractErrorMessage(error?.response?.data)
+      });
+      onAddAlert({ title: errorMessage, variant: 'danger', id: helpers.generateId() });
+    },
+    [onAddAlert, t]
+  );
+
+  const editSources = useCallback(
+    async (payload: SourceType) => {
+      try {
+        const response = await apiCall(payload);
+        callbackSuccess(response);
+      } catch (error) {
+        if (isAxiosError(error)) {
+          return callbackError(error, payload.name);
+        }
+        if (!helpers.TEST_MODE) {
+          console.error(error);
+        }
+      }
+    },
+    [apiCall, callbackSuccess, callbackError]
+  );
+
+  return {
+    apiCall,
+    callbackError,
+    callbackSuccess,
+    editSources
   };
+};
 
-  /**
-   * Deletes several selected sources based on user interaction.
-   * Add your logic to handle the deletion of selected items within this function.
-   */
-  const onDeleteSelectedSources = () => {
-    const selectedItems = [];
-    console.log('Deleting selected sources:', selectedItems);
+const useShowConnectionsApi = (setConnectionsData: (data: any) => void) => {
```

Comment:
I agree with proposal you brought above, I would like to suggest us moving those hooks there in the scans api PR, solely because I have ongoing work on that PR and I want to avoid trouible while rebasing. As soon as we merge sources work, I will have a commit moving these pieces there before I open the scan PR. would that be ok with you ?

Reply:
That works for me!